deps/first/File/src/File.cpp

```diff
@@ -22,8 +24,57 @@ namespace RC::File
         return construct_handle(file_path_and_name, open_properties);
     }
 
+    // BOM sequences
+    const std::string BOM_UTF8 = "\xEF\xBB\xBF";
+    const std::wstring BOM_UTF16_LE = L"\xFEFF";
+    const std::wstring BOM_UTF16_BE = L"\xFFFE";
+
+    // Function to check for BOM and reopen the file as a wide stream
+    auto open_file_skip_BOM(const std::wstring& filename) -> StreamType
+    {
+        std::ifstream file(filename, std::ios::binary);
+        if (!file.is_open())
+        {
+            throw std::runtime_error("Failed to open the file.");
+        }
+
+        char bom[3] = {0};
+        file.read(bom, 3);
+        if (std::string(bom, 3) == BOM_UTF8)
+        {
+            // UTF-8 BOM detected
+            file.close();
+            StreamType wfile(filename);
+            wfile.seekg(3, std::ios::beg); // Skip BOM
+            return wfile;
+        }
+        else
+        {
+            // Check for wide BOM
+            wchar_t wbom;
+            file.seekg(0, std::ios::beg); // Rewind to start
+            file.read(reinterpret_cast<char*>(&wbom), sizeof(wchar_t));
+            file.close();
+            if (wbom == BOM_UTF16_LE[0] || wbom == BOM_UTF16_BE[0])
+            {
+                StreamType wfile(filename);
+#pragma warning(disable : 4996)
+                wfile.imbue(std::locale(wfile.getloc(), new std::codecvt_utf8_utf16<wchar_t>));
```

Comment:
Is this a memory leak ?
There's usage of `new` without a corresponding use of `delete`.